custom_model_runner/datarobot_drum/resource/components/Python/r_fit/fit.R

```diff
@@ -24,6 +24,69 @@ init <- function(code_dir) {
 }
 
 
+load_data <- function(input_filename){
+    tmp = readChar(input_filename, file.info(input_filename)$size)
+    data <- read.csv(text=gsub("\r","", tmp, fixed=TRUE))
+    data
+}
+
+
+process_data <- function(input_filename, target_filename, target_name, num_rows){
+    set.seed(1)
+    # read X
+    df <- load_data(input_filename)
+    # set num_rows
+    if (num_rows == 'ALL'){
+        num_rows <- nrow(df)
+    } else {
+        num_rows <- as.integer(a)
+    }
+    # handle target
+    if (!is.null(target_filename) || !is.null(target_name)){
+        if (!is.null(target_filename)){
+            y_unsampled = load_data(target_filename)
+            stopifnot( length(colnames(y_unsampled)) == 1)
+            stopifnot( nrow(df) == nrow(y_unsampled))
+            df <- cbind(df, y)
+            target_name <- colnames(y)
+        }
+    df <-  df[!(is.na(df[target_name])) , ]
+    X <-  df[,!(names(df) %in% c(target_name))]
+    X <- X[sample(nrow(X), size=num_rows, replace=TRUE ), ]
+
+    y <- df[,target_name, drop=FALSE]
+    y <- y[sample(nrow(y), size=num_rows, replace=TRUE),]
+
+    } else {
+
+        y <- NULL
+        X <- df[sample(nrow(df), size=num_rows, replace=TRUE ), ]
+
+    }
+
+    return (list('X' = X, 'y' = y, 'num_rows' = num_rows))
+
+}
+
+
+process_weights <- function(X, weights_filename, weights, num_rows){
+  if (!is.null(weights_filename)){
+    set.seed(1)
```

Comment:
whats this do?

Reply:
it looks like rather than passing the random seed to individual functions, you set.seed in its own call in R - I think actually I can put this once at the top...will confirm that